python/pypto/jit/decorator.py

```
@@ -392,49 +392,58 @@ def _extract_call_args_for_dep(
-    entry param/dim pair should also be dynamic.
+    ``call_args`` is the return value of ``_extract_call_args_for_dep``: a
+    plain list of names for positional calls, or a list of (param, arg)
+    pairs for keyword-only calls. Both forms collapse to the same dict.
+    ``_extract_call_args_for_dep`` only emits keyword tuples whose first
+    element is a real ``str`` (it skips ``**kwargs`` entries), so the cast
+    in the keyword branch is sound.
+    """
+    if call_args and isinstance(call_args[0], tuple):
+        return dict(cast("list[tuple[str, str | None]]", call_args))
+    pos_args = cast("list[str | None]", call_args)
+    return dict(zip(dep_param_names, pos_args))
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)